### PR TITLE
[v4.0-rhel] Bump github.com/containerd/containerd to v1.5.18

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Bump containerd to v1.5.18 to address CVE-2023-25173 and the Jira card:

https://issues.redhat.com/browse/RHEL-13490

[NO NEW TESTS NEEDED]

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@vrothberg if you could PTAL, I'd appreciate it.  In this branch, containerd was only in go.sum.  After the vendor bump with `go get`, it's now in go.mod.  I think that's OK, but would like a confirm.

I'm also leary about doing this as this branch is not the newest, but we're supposed to make a best effort given the CVE.

**Comment by Luap99** [comment]
What is the point of this? We do not use containerd to create containers so we cannot vulnerable by this, the changed code for this fix is not even vendored here so we 100% do not use it and I would just close it as non applicable to us.

We do had the same issue but they were already fixed as https://github.com/advisories/GHSA-4wjj-jwc9-2x96 and https://github.com/advisories/GHSA-fjm8-m7m6-2fjp

**Review by vrothberg — COMMENTED** [review]
Assuming it's just to silence some static analyzer, LGTM

**Review by Luap99 — CHANGES_REQUESTED** [review]
This cannot be merged like this, looks like the whole vendor process got broken.

But I still think we should say we are not vulnerable and close this.

File: vendor/github.com/opencontainers/runc/libcontainer/cgroups/file.go [hunk 1]
```diff
@@ -76,16 +76,16 @@ var (
 	// TestMode is set to true by unit tests that need "fake" cgroupfs.
 	TestMode bool
 
-	cgroupRootHandle *os.File
```
[the remark this answers is not in the document]
Comment (vrothberg) [reply vendor/github.com/opencontainers/runc/libcontainer/cgroups/file.go:79]: Probably best to clone runc somewhere or create a branch upstream at runc.

[the remark this answers is not in the document]
Comment (mheon) [reply vendor/github.com/opencontainers/runc/libcontainer/cgroups/file.go:79]: https://github.com/containers/podman/pull/21851

**Comment by TomSweeneyRedHat** [comment]
@Luap99 and @mheon, based on the above discussions and the fact that we've made other changes to the go.mod in this branch, I decided to wipe the original commit and redo it. This seems a lot more reasonable now. Only 26 files are touched rather than almost 200, and the go.mod looks a lot better. IDK if I messed up the original vendor, but I wouldn't bet against it.

I spent a little time looking to see if we actually called any of the containerd code that was problematic, but after 30 minutes, I couldn't confirm or deny, so I took the chicken's way out and did this over.

**Comment by Luap99** [comment]
We do not vendor any of the effected code, and we do not use any container creation code from containerd so we cannot be vulnerable by this containerd vulnerability as mentioned above. The linked issue even link our podman/buildah CVE that we fixed long ago already.

And if you look at the fix in containerd (https://github.com/containerd/containerd/commit/a62c38bf2173faa813018939710fc8491e4f7dba) there you can see the fix is in the oci package which is not even vendored by us as mentioned above.
```
$ ls vendor/github.com/containerd/containerd/oci
ls: cannot access 'vendor/github.com/containerd/containerd/oci': No such file or directory
```

**Comment by TomSweeneyRedHat** [comment]
@Luap99 thanks for looking into that.  I just double checked using your method and my standard grep, and I'm not seeing it either.  We also have to worry about `github.com/containerd/containerd/cri/server` but that also is not present.  I'll close this out and will backout the remaining cards.

**Pull request closed without merging**